/*======================================================================================================================
 * editors/HopEditor.h is part of Brewken, and is copyright the following authors 2009-2023:
 *   • Jeff Bailey <skydvr38@verizon.net>
 *   • Mik Firestone <mikfire@gmail.com>
 *   • Philip Greggory Lee <rocketman768@gmail.com>
 *
 * Brewken is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * Brewken is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
 * details.
 *
 * You should have received a copy of the GNU General Public License along with this program.  If not, see
 * <http://www.gnu.org/licenses/>.
 =====================================================================================================================*/
#ifndef EDITORS_HOPEDITOR_H
#define EDITORS_HOPEDITOR_H
#pragma once

#include "ui_hopEditor.h"
#include <QMetaProperty>
#include <QVariant>

#include "editors/EditorBase.h"

// Forward declarations.
class Hop;

/*!
 * \class HopEditor
 *
 * \brief View/controller class for creating and editing Hops.
 *
 *        See comment on EditorBase::connectSignalsAndSlots for why we need to have \c public, not \c private
 *        inheritance from the Ui base.
 */
class HopEditor : public QDialog, public Ui::hopEditor, public EditorBase<HopEditor, Hop> {
   Q_OBJECT

   EDITOR_COMMON_DECL(Hop)
};

#endif
